shared/src/simulation/behaviors/manage-patient-transport-to-hospital.ts

```diff
@@ -0,0 +1,513 @@
+
+                            if (
+                                Object.values(
+                                    behaviorState.patientsExpectedInRegions
+                                ).every(
+                                    (patientCount) =>
+                                        patientCount[event.patientCategory] ===
+                                        0
+                                )
+                            ) {
+                                addActivity(
+                                    simulatedRegion,
+                                    DelayEventActivityState.create(
+                                        nextUUID(draftState),
+                                        PatientCategoryTransferToHospitalFinishedEvent.create(
+                                            event.patientCategory,
+                                            false
+                                        ),
+                                        draftState.currentTime
+                                    )
+                                );
+                            }
+                        }
+                    }
+                    break;
+
+                case 'askForPatientDataEvent':
+                    {
+                        // only react if this event is meant for this behavior
+
+                        if (event.behaviorId !== behaviorState.id) {
+                            break;
+                        }
+
+                        // if it manages its own simulated region initiate a patient count
+
+                        if (
+                            behaviorState.simulatedRegionsToManage[
+                                simulatedRegion.id
+                            ]
+                        ) {
+                            addActivity(
+                                simulatedRegion,
+                                CountPatientsActivityState.create(
+                                    nextUUID(draftState)
+                                )
+                            );
+                        }
+
+                        if (
+                            Object.keys(
+                                behaviorState.simulatedRegionsToManage
+                            ).some(
+                                (simulatedRegionId) =>
+                                    simulatedRegionId !== simulatedRegion.id
+                            )
+                        ) {
+                            publishRadiogram(
+                                draftState,
+                                cloneDeepMutable(
+                                    NewPatientDataRequestedRadiogram.create(
+                                        nextUUID(draftState),
+                                        simulatedRegion.id,
+                                        RadiogramUnpublishedStatus.create()
+                                    )
+                                )
+                            );
```

Comment:
Please use a public radiogram activity since behaviors are not supposed to manipulate the state besides their own behavior state and state manipulations necessary to create/configure activities.